service/src/main/java/bio/terra/landingzone/stairway/flight/delete/DeleteLandingZoneResourcesStep.java

```diff
@@ -97,7 +119,7 @@ private DeletedLandingZone deleteLandingZoneResources(
       throws LandingZoneRuleDeleteException {
     if (isAttached) {
       logger.info("Landing zone {} was attached, skipping Azure resource deletion", landingZoneId);
-      return new DeletedLandingZone(landingZoneId, Collections.emptyList(), billingProfileId);
+      return DeletedLandingZone.emptyLandingZone(landingZoneId, billingProfileId);
```

Comment:
I'm a bit confused about this codepath. If the landing zone is attached, why are we deleting it at all?

Reply:
Oh, that's what "attached" refers to. Got it!